azure-toolkit-libs/azure-toolkit-applicationinsights-lib/src/main/java/com/microsoft/azure/toolkit/lib/applicationinsights/ApplicationInsights.java

```diff
@@ -0,0 +1,131 @@
+import com.microsoft.azure.toolkit.lib.common.cache.Cacheable;
+import com.microsoft.azure.toolkit.lib.common.model.Region;
+import com.microsoft.azure.toolkit.lib.common.model.Subscription;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class ApplicationInsights extends SubscriptionScoped<ApplicationInsights> implements AzureService {
+    public ApplicationInsights() { // for SPI
+        super(ApplicationInsights::new);
+    }
+
+    private ApplicationInsights(@Nonnull final List<Subscription> subscriptions) {
+        super(ApplicationInsights::new, subscriptions);
+    }
+
+    public boolean exists(@Nonnull String resourceGroup, @Nonnull String name) {
+        return exists(getDefaultSubscription().getId(), resourceGroup, name);
+    }
+
+    public boolean exists(@Nonnull String subscriptionId, @Nonnull String resourceGroup, @Nonnull String name) {
+        try {
+            return get(subscriptionId, resourceGroup, name) != null;
+        } catch (ManagementException e) {
+            // SDK will throw exception when resource not founded
+            return false;
+        }
+    }
+
+    public ApplicationInsightsEntity get(@Nonnull String resourceGroup, @Nonnull String name) {
+        return get(getDefaultSubscription().getId(), resourceGroup, name);
+    }
+
+    public ApplicationInsightsEntity get(@Nonnull String subscriptionId, @Nonnull String resourceGroup, @Nonnull String name) {
+        return Optional.ofNullable(getApplicationInsightsManager(subscriptionId).components().getByResourceGroup(resourceGroup, name))
+                .map(ApplicationInsights::getFromApplicationInsightsComponent)
+                .orElse(null);
+    }
+
+    public ApplicationInsightsEntity create(@Nonnull String resourceGroup, @Nonnull Region region, @Nonnull String name) {
+        return create(getDefaultSubscription().getId(), resourceGroup, region, name);
+    }
+
+    /**
+     * Create application insights instance with kind:web and applicationType:web
+     *
+     * @param subscriptionId id for subscription to create the resource
+     * @param resourceGroup  existing resource to create the resource
+     * @param region         region to resource to create
+     * @param name           name for resource to create
+     * @return ApplicationInsightsEntity for created resource
+     */
+    public ApplicationInsightsEntity create(@Nonnull String subscriptionId, @Nonnull String resourceGroup, @Nonnull Region region, @Nonnull String name) {
+        final ApplicationInsightsComponent component = getApplicationInsightsManager(subscriptionId).components().define(name)
+                .withRegion(region.getName()).withExistingResourceGroup(resourceGroup).withKind("web").withApplicationType(ApplicationType.WEB).create();
+        return getFromApplicationInsightsComponent(component);
+    }
+
+    public List<ApplicationInsightsEntity> list() {
```

Comment:
eturn Flux.fromIterable(getSubscriptions()).parallel()

Reply:
Can not using Flux here as application insights SDK did not provide async interface